pkgs/by-name/ma/material-maker/package.nix

```diff
@@ -0,0 +1,116 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+
+  # build deps
+  autoPatchelfHook,
+  copyDesktopItems,
+  makeDesktopItem,
+
+  # runtime deps
+  libGL,
+  libGLU,
+  libX11,
+  libXcursor,
+  libXext,
+  libXfixes,
+  libXi,
+  libXinerama,
+  libXrandr,
+  libXrender,
+}:
+let
+  pname = "material-maker";
+  version = "1.3";
+  versionUnderscore = "${lib.replaceStrings [ "." ] [ "_" ] version}";
+
+  sources = {
+    linux = fetchzip {
+      url = "https://github.com/RodZill4/${pname}/releases/download/${version}/material_maker_${versionUnderscore}_linux.tar.gz";
+      hash = "sha256-WEu5gVfnswB5zYzu3leOL+hKOBzJbn48gHQKshlfOh4=";
+    };
+    darwin = fetchzip {
+      url = "https://github.com/RodZill4/${pname}/releases/download/${version}/material_maker_${versionUnderscore}.dmg";
+      hash = "07isbwcv42klh0y7wvn93hw4ms1girbxxvlcrmwh1cz7ay0vjjsq";
+    };
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit pname version;
+
+  src =
+    if stdenv.isLinux && stdenv.isx86_64 then
+      sources.linux
+    else if stdenv.isDarwin then
+      sources.darwin
+    else
+      throw "unsupported platform";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    libGL
+    libGLU
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+  ];
+
+  desktopItems = makeDesktopItem {
+    name = "material-maker";
+    exec = "material-maker";
+    icon = "material-maker";
+    genericName = "Procedural texture generation and 3D model painting tool";
+    desktopName = "Material Maker";
+    comment = finalAttrs.meta.description;
+    terminal = false;
+    categories = [
+      "Graphics"
+      "3DGraphics"
+    ];
+    keywords = [ "material" ];
+  };
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      install -D -m 755 $src/material_maker.x86_64 $out/libexec/material-maker
+      install -D -m 644 $src/material_maker.pck $out/libexec/material-maker.pck
+
+      for dir in examples library nodes doc environments meshes export; do
+        cp -R $src/$dir $out/libexec
+      done
+
+      install -D -m 644 $src/doc/_static/icon.png -t $out/share/icons/hicolor/256x256/apps/
+
+      mkdir -p $out/bin
+      ln -s $out/libexec/material-maker $out/bin/material-maker
+    ''
+    + lib.optionalString stdenv.isDarwin ''
```

Comment:
```suggestion
  stdenvNoCC,
  fetchzip,

  # build deps
  autoPatchelfHook,
  copyDesktopItems,
  makeDesktopItem,

  # runtime deps
  libGL,
  libGLU,
  libX11,
  libXcursor,
  libXext,
  libXfixes,
  libXi,
  libXinerama,
  libXrandr,
  libXrender,
}:
let
  pname = "material-maker";
  version = "1.3";
  versionUnderscore = "${lib.replaceStrings [ "." ] [ "_" ] version}";

  sources = {
    linux = fetchzip {
      url = "https://github.com/RodZill4/${pname}/releases/download/${version}/material_maker_${versionUnderscore}_linux.tar.gz";
      hash = "sha256-WEu5gVfnswB5zYzu3leOL+hKOBzJbn48gHQKshlfOh4=";
    };
    darwin = fetchzip {
      url = "https://github.com/RodZill4/${pname}/releases/download/${version}/material_maker_${versionUnderscore}.dmg";
      hash = "07isbwcv42klh0y7wvn93hw4ms1girbxxvlcrmwh1cz7ay0vjjsq";
    };
  };
in
stdenvNoCC.mkDerivation (finalAttrs: {
  inherit pname version;

  src =
    if stdenvNoCC.isLinux && stdenvNoCC.isx86_64 then
      sources.linux
    else if stdenvNoCC.isDarwin then
      sources.darwin
    else
      throw "unsupported platform";

  nativeBuildInputs = [
    autoPatchelfHook
    copyDesktopItems
  ];

  buildInputs = [
    libGL
    libGLU
    libX11
    libXcursor
    libXext
    libXfixes
    libXi
    libXinerama
    libXrandr
    libXrender
  ];

  desktopItems = makeDesktopItem {
    name = "material-maker";
    exec = "material-maker";
    icon = "material-maker";
    genericName = "Procedural texture generation and 3D model painting tool";
    desktopName = "Material Maker";
    comment = finalAttrs.meta.description;
    terminal = false;
    categories = [
      "Graphics"
      "3DGraphics"
    ];
    keywords = [ "material" ];
  };

  installPhase =
    ''
      runHook preInstall
    ''
    + lib.optionalString stdenvNoCC.isLinux ''
      install -D -m 755 $src/material_maker.x86_64 $out/libexec/material-maker
      install -D -m 644 $src/material_maker.pck $out/libexec/material-maker.pck

      for dir in examples library nodes doc environments meshes export; do
        cp -R $src/$dir $out/libexec
      done

      install -D -m 644 $src/doc/_static/icon.png -t $out/share/icons/hicolor/256x256/apps/

      mkdir -p $out/bin
      ln -s $out/libexec/material-maker $out/bin/material-maker
    ''
    + lib.optionalString stdenvNoCC.isDarwin ''
```

Since no C compiler is needed